### DISCUSSION TITLE
Unexpected behaviour of app router when deployed to vercel

### DISCUSSION THREAD

**Discussion by Honzoraptor31415** [vercel/next.js#69938] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/Honzoraptor31415/PortfolioV2/blob/main/src/app/not-found.tsx

### To Reproduce

1. Deploy the mentioned app to vercel
2. Go to a route that doesn't exist (ex. /some-random-route-ahhh123)
3. Witness

### Current vs. Expected behavior

##Current behaviour

Currentely, when I try to run my app localy and go to a route that doesn't exist, my 404 page works normally. When I try to go to /404, it also works normally and when I go to /500 it also displays the 404 error page, which is how it should work in production as well, but when I try to deploy the app to Vercel and go to a route that doesn't exist, it shows me the homepage, when I go to /404, it shows me the 404 error page and when I go to /500, it shows me the preconfugured Next.js error page.

Also in the build logs on Vercel I noticed, that it generates /404 and a /500 static routes:
![image](https://github.com/user-attachments/assets/ff38e479-d72a-4686-ad5e-26d55cfa5fac)
Which is definitely not something that should be happening am I right?

##Expected behaviour

The routing should work the same in production like it works when running the app locally.

### Provide environment information

```bash
Next.js version: 14.2.3
OS: (doesn't apply I think) Windows 11
Node.js version: (on my local machine - also doesn't apply) v21.1.0
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

### Additional context

In case you were wondering, **yes**, I do have Next.js selected in the framework preset section:
![image](https://github.com/user-attachments/assets/1ccbac08-cc0a-4dd8-8cdd-175376f33ddd)

**Comment by samcx** [comment, marked as answer]
@Honzoraptor31415 This is expected. When you build locally, Next.js is platform agnostic, so going to `/500` is just a unknown path, so it shows your `/404` page. On Vercel, we also build our (I believe all Vercel deploys have this `/500` page) `/500` page for you.

I converted this to a discussion because this is not an issue with Next.js!

**Reply by Honzoraptor31415** [reply]
Ok, but what about the 404 page not functioning properly?

**Reply by samcx** [reply]
@Honzoraptor31415 My apologies I forgot that point.

If you remove your `vercel.json`, I believe the `404` functions properly. With Next.js on Vercel, you should not use a `vercel.json` for routing—you should use `next.config` instead.

**Reply by Honzoraptor31415** [reply]
> @Honzoraptor31415 My apologies I forgot that point.
> 
> If you remove your `vercel.json`, I believe the `404` functions properly. With Next.js on Vercel, you should not use a `vercel.json` for routing—you should use `next.config` instead.

Thanks a lot!
